test/hotspot/jtreg/compiler/inlining/TestNullAtCallAfterLateInline.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2023, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * bug 8318826
+ * @summary C2: "Bad graph detected in build_loop_late" with incremental inlining
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -XX:+AlwaysIncrementalInline -XX:-BackgroundCompilation TestNullAtCallAfterLateInline
```

Comment:
`AlwaysIncrementalInline` is a debug option.

Reply:
May be we should make it diagnostic and add it to stress testing flags.
Or, as Tobias pointed, `@requires` for debug VM is needed here.

Reply:
Thanks for looking at this. I added `-XX:+IgnoreUnrecognizedVMOptions` so the test case is run in any case.